File: wordCount.js
class WordFrequenciesModel {
    constructor(pathToFile) {
        /** Define a estrutura principal da classe, um Map do tipo:
         * {'palavra':frequencia, ...}
         */
        this.termFrequency = new Map();

        /** Realiza o parser do arquivo stop_words e põe as palavras em um array. */
        this.stopWords = readFile('stop_words.txt').split("\r\n");
        this.stopWords = Array.from(new Set(this.stopWords));
        
        this.update(pathToFile);
    }

    /**
     * Conta com qual frequência as palavras especificadas em stop_words.txt 
     aparecem em um determinado arquivo e as armazena na variável termFrequency.
     * 
     * @param {string} pathToFile O arquivo no qual as palavras serão procuradas.
     */
    update(pathToFile) {
        /** Tenta ler e realizar o parser do arquivo desejado. */
        try {
            const file = readFile(pathToFile).split("\r\n");
            const words = [];

            file.forEach(word => {
                if (word.match("[a-z]{2,}")) {
                    words.push(word);
                }
            })

            /** Limpa o Map declarando-o novamente. */
            this.termFrequency = new Map();

            /** Realiza a contagem da frequência das palavras. */
            words.forEach(word => {
                if (word in this.termFrequency) {
                    this.termFrequency[word]++;    
                } else if (!(this.stopWords.includes(word))) {
                    this.termFrequency[word] = 1;
                }
            })
        } catch (error) {
            console.log("File not found!");
        }
    } 
}


class WordFrequenciesView {
    constructor(model) {
        this._model = model; 
    }

    /** Exibe as informações da contagem na tela. */
    render() {
        for (const word in this._model.termFrequency) {
            console.log(`${word} -> ${this._model.termFrequency[word]}`);
        }
    }
}


class WordFrequencyController {
    constructor(model, view) {
        this._model = model;
        this._view = view;

        view.render();
    }

    /**
     * Executa um loop de leitura para arquivos informados no console.
     */
    async run() {
        while (true) {         
            const filename = await askQuestion("Next file: ");
            this._model.update(filename);
            this._view.render();
        }
    }
}


// Funções auxiliares --------------------------------- //

/**
 * Lê um arquivo e põe seu conteúdo em uma string.
 * 
 * @param {string} pathtoFile O caminho para o arquivo a ser lido.
 * @return Uma string contendo com o conteúdo do arquivo.
 */
const readFile = function(pathToFile) {
    const fs = require("fs");
    return fs.readFileSync(pathToFile, 'utf8')
}

/**
 * Exibe uma pergunta no console e coleta a resposta do usuário.
 * 
 * @param {string} question A pergunta a ser exibida.
 * @return Uma promise resolvida (string) contendo o input de resposta do usuário.
 */
function askQuestion(question) {
    const readline = require('readline');
    
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout,
    });

    return new Promise(resolve => rl.question(question, ans => {
        rl.close();
        resolve(ans);
    }))
}

// ---------------------------------------------------- //

m = new WordFrequenciesModel('./words.txt');
v = new WordFrequenciesView(m);
c = new WordFrequencyController(m, v);
c.run();

File: wordCount.py
#!/usr/bin/env python
  
import sys, re, operator, collections

class WordFrequenciesModel:
    ''' 
    Models the data. In this case, we're only interested
    6 in words and their frequencies as an end result 
    '''

    # Dicionário que contém os termos (palavras) e sua frequência.
    term_frequency = {}
    stopwords = set(open('./stop_words.txt').read().split(','))

    def __init__(self, path_to_file):
        self.update(path_to_file)


    def update(self, path_to_file):
        '''
        Função que preenche o dicionário term_frequency.
        '''
        try:
            # Filtra todas as palavras com mais de 1 letra e armazena em uma lista.
            words = re.findall('[a-z]{2,}', open(path_to_file).read().lower())

            # Preenche o dicionário de frequência.
            # {'palavra1':2, 'palavra2':1, 'palavra3':5, ...}
            self.term_frequency = collections.Counter(w for w in words if w not in self.stopwords)

        except IOError:
            print("File not found")


class WordFrequenciesView:
    _model = None

    def __init__(self, model):
        self._model = model

    def render(self):
        # Ordena as palavras.
        sorted_term_frequency = sorted(self._model.term_frequency.items(), key=operator.itemgetter(1), reverse=True)
        
        # Itera pelo dicionário e exibe os dados.
        for (word, frequency) in sorted_term_frequency[0:25]:
            print(word, '-', frequency)


class WordFrequencyController:
    def __init__(self, model, view):
        self._model, self._view = model, view
        view.render()

    def run(self):
        while True:
            print("Next file: ")
            sys.stdout.flush()
            filename = sys.stdin.readline().strip()
            self._model.update(filename)    
            self._view.render()


m = WordFrequenciesModel(sys.argv[0])
v = WordFrequenciesView(m)
c = WordFrequencyController(m, v)
c.run()

# Primeiro o programa lisa todas as ocorrências do arquivo principal